### PR TITLE
Fix compilation error

### PR DESCRIPTION
**Pull request by pruivo**

### CODE REVIEW

**Comment by pruivo** [comment]
Let's wait for the tests to run.

**Comment by pruivo** [comment]
The Protostream pull request caused a conflict with https://github.com/keycloak/keycloak/pull/30381 and the main branch is not compiling anymore.

I'm surprised the test suite didn't fail with https://github.com/keycloak/keycloak/pull/30381 because the events are cast to `RealmCacheInvalidationEvent`. See:

https://github.com/keycloak/keycloak/blob/f44c5b64db67d125ccd4f6e75b63d440697e0e03/model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmCacheManager.java#L135

I've created new events. Can @vramik @pedroigor review this PR, please? Thanks!

**Comment by ahus1** [comment]
@pruivo - I talked to @pedroigor and he would be ok to merge this now to fix the broken build and then discuss later. He'll be in contact with you.

File: model/storage-private/src/main/java/org/keycloak/models/cache/CacheRealmProvider.java [hunk 1]
```diff
@@ -39,5 +39,7 @@ public interface CacheRealmProvider extends RealmProvider, ClientProvider, Clien
     void registerRoleInvalidation(String id, String roleName, String roleContainerId);
 
     void registerGroupInvalidation(String id);
-    void registerInvalidation(String id);
```
Comment (pedroigor) [inline model/storage-private/src/main/java/org/keycloak/models/cache/CacheRealmProvider.java:42]: One of the reasons we added this method is to allow register invalidations on the realm cache for whatever entry we want. Therefore, avoid having methods like those below that are specific to organizations.

Can we keep the previous contract?

Comment (pruivo) [reply model/storage-private/src/main/java/org/keycloak/models/cache/CacheRealmProvider.java:42]: Yes, I can add a "generic" event and keep the old contract :+1:

**Comment by pruivo** [comment]
Thanks @ahus1 @pedroigor. I've updated the PR.

**Review by pedroigor — APPROVED** [review]
Thanks!

**Comment by pruivo** [comment]
ps. didn't run the test suite locally, better wait for GHA report :+1:

**Pull request merged**